packages/core/contracts/financial-templates/common/FundingRateApplier.sol

```diff
@@ -68,22 +81,88 @@ abstract contract FundingRateApplier is Testable, Lockable {
      *              MODIFIERS               *
      ****************************************/
 
-    modifier updateFundingRate {
-        _applyEffectiveFundingRate();
+    modifier fees override {
+        applyFundingRate();
+        _;
+    }
+
+    modifier regularFees {
+        payRegularFees();
         _;
     }
 
     /**
      * @notice Constructs the FundingRateApplier contract. Called by child contracts.
+     * @param _fundingRateIdentifier identifier that tracks the funding rate of this contract.
+     * @param _collateralAddress address of the collateral token.
      * @param _finderAddress Finder used to discover financial-product-related contracts.
+     * @param _configStoreAddress address of the remote configuration store managed by an external owner.
+     * @param _timerAddress address of the timer contract in test envs, otherwise 0x0.
      */
-    constructor(address _finderAddress) public {
-        finder = FinderInterface(_finderAddress);
+    constructor(
+        bytes32 _fundingRateIdentifier,
+        address _collateralAddress,
+        address _finderAddress,
+        address _configStoreAddress,
+        address _timerAddress
+    ) public FeePayer(_collateralAddress, _finderAddress, _timerAddress) {
+        uint256 currentTime = getCurrentTime();
+        fundingRate.updateTime = currentTime;
+        fundingRate.applicationTime = currentTime;
+
+        // Seed the cumulative multiplier as 1, from which it will be scaled as funding rates are applied over time.
+        fundingRate.cumulativeMultiplier = FixedPoint.fromUnscaledUint(1);
         emergencyShutdownTimestamp = 0;
-        lastUpdateTime = getCurrentTime();
 
-        // Seed the initial funding rate in the cumulativeFundingRateMultiplier 1.
-        cumulativeFundingRateMultiplier = FixedPoint.fromUnscaledUint(1);
+        fundingRate.identifier = _fundingRateIdentifier;
+        configStore = ConfigStoreInterface(_configStoreAddress);
```

Comment:
Should we do any checking that _configStoreAddress is a valid contract?

Reply:
Prehaps a good idea to verify it here in the constructor. If it's not the perp is pretty broken

Reply:
Yeah there are no fallback values or anything if the ConfigStore reverts on `getConfigSettings`

Reply:
How should we check that? By doing a test `getConfigSettings` call or doing an `isContract` check?

My only minor objection to doing checks in the contract is that we decided not to do similar checks for the synthetic token to save bytecode.

Reply:
> My only minor objection to doing checks in the contract is that we decided not to do similar checks for the synthetic token to save bytecode.

This is a good point, and it would be consistent to maintain a "buyer beware" stance here with this method (i.e. don't interact with a Perp that uses a bad ConfigStore).

Reply:
Yeah, like the synthetic token, if you launch this contract through the contract creator (and that's the only way it's approved), the config store is guaranteed to be valid.